crates/wasmtime/src/func.rs

```diff
@@ -193,17 +193,26 @@ macro_rules! getters {
                         unsafe extern "C" fn(
                             *mut VMContext,
                             *mut VMContext,
-                            $($args,)*
-                        ) -> R,
+                            $( $args::Abi, )*
+                        ) -> R::Abi,
                     >(anyfunc.as_ref().func_ptr.as_ptr());
+
                     let mut ret = None;
-                    $(let $args = $args.into_abi();)*
+
+                    let weak_store = instance.store.weak();
+                    let weak_store = WeakStore(&weak_store);
+
+                    $( let $args = $args.into_abi_for_arg(weak_store); )*
```

Comment:
Looking again at this, I think this location also needs a check for store compatibility because the `$args` are arbitrarily entering here from the host and we're passing them off to wasm below.

Reply:
good eye -- also nice that `compatible_with_store` is its own method, rather than a fused operation with `into_abi_for_arg`, because it is super simple to reuse here.